src/interactive-window/interactiveWindow.ts

```diff
@@ -309,21 +334,50 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
             .ignoreErrors();
     }
 
+    private deleteSysInfoCell(cellPromise: Promise<NotebookCell | undefined>) {
+        this._insertSysInfoPromise = undefined;
+        cellPromise
+            .then((cell) =>
+                chainWithPendingUpdates(this._notebookDocument!, (edit) => {
+                    if (cell !== undefined && cell.index >= 0) {
+                        if (
+                            cell.kind === NotebookCellKind.Markup &&
+                            cell.metadata.isInteractiveWindowMessageCell &&
+                            cell.metadata.isPlaceholder
+                        ) {
+                            edit.replaceNotebookCells(
+                                cell.notebook.uri,
+                                new NotebookRange(cell.index, cell.index + 1),
+                                []
+                            );
+                            return;
+                        }
+                    }
+                })
+            )
+            .ignoreErrors();
+    }
+
     private finishSysInfoMessage(
-        kernelOrError: IKernel | Error,
+        kernel: IKernel,
         cellPromise: Promise<NotebookCell | undefined>,
         reason: SysInfoReason
     ) {
-        const kernelName =
-            'info' in kernelOrError ? kernelOrError.kernelConnectionMetadata.interpreter?.displayName : '';
-        const kernelInfo =
-            'info' in kernelOrError && kernelOrError.info?.status === 'ok' ? kernelOrError.info : undefined;
-        const banner = kernelInfo ? kernelInfo.banner.split('\n').join('  \n') : kernelOrError.toString();
+        const kernelName = 'info' in kernel ? kernel.kernelConnectionMetadata.interpreter?.displayName : '';
```

Comment:
Does the kernelName = 'info' make sense here? It's not looking at the kernel.info like below. Maybe kernelConnectionMetadata?

Reply:
I'm going to remove this and have a separate method that deals with errors, feels weird to look for `info` or something else to figure out whether its kernel or an error.

Reply:
Fixed this issue in this PR https://github.com/microsoft/vscode-jupyter/pull/9483
Basically split into two distinct methods, no overloads